### DISCUSSION TITLE
Share environment variables across multiple runners

### DISCUSSION THREAD

**Discussion by JavierSegoviaCordoba** [community/community#24978] [Actions]
<p>Use case: I need to run a previous job/step in a Linux machine to set an ID that will be used in the next jobs/steps that will run on a matrix of Windows and macOS machines).</p>
<p>For example:</p>
<pre><code class="lang-auto">steps:
  - name: Set the value
    id: step_one
    if: ${{ matrix.os == 'ubuntu-latest' }}
    run: |
        echo "action_state=yellow" &gt;&gt; $GITHUB_ENV
  - name: Use the value
    id: step_two
    run: |
        echo "${{ env.action_state }}" # 'yellow' output in macOS and Windows?
</code></pre>
<p>I don’t care if this can be possible with multiple steps or multiple jobs, if with multiple jobs it is possible I would like to know how.</p>

**Comment by chrispat** [comment, marked as answer]
<p>To run on a different OS you have to use a different job.  All steps in a given job are executed by a single runner.  You can pass data between jobs using job outputs <a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idoutputs" rel="noopener nofollow ugc">Workflow syntax for GitHub Actions - GitHub Docs</a>.</p>
